src/library/scala/concurrent/Promise.scala

```diff
@@ -17,6 +17,10 @@ import scala.util.{ Try, Success, Failure }
 /** Promise is an object which can be completed with a value or failed
  *  with an exception.
  *
+ *  A promise should always be completed, whether for success or failure. 
+ *  Failing to complete a promise can lead to memory being retained 
+ *  for any associated future.
+ *
```

Comment:
How about:

> A promise should always eventually be completed, whether for success or failure, in order to avoid unintended resource retention for any associated `Future`'s callbacks or transformations.

Reply:
Done.

Reply:
@jroper You OK with this wording?